### PR TITLE
rename gettext macro

### PR DESCRIPTION
**Pull request by arogge**
#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
~~Check backport line~~
~~Required backport PRs have been created~~

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
~~Required documentation changes are present and part of the PR~~

### CODE REVIEW

File: core/src/dird/ua_update.cc [hunk 1]
```diff
@@ -613,41 +615,41 @@ static bool UpdateVolume(UaContext* ua)
...
+    AddPrompt(ua, T_("Volume Files"));               /* 9 */
+    AddPrompt(ua, T_("Pool"));                       /* 10 */
+    AddPrompt(ua, T_("Volume from Pool"));           /* 11 */
+    AddPrompt(ua, T_("All Volumes from Pool"));      /* 12 */
+    AddPrompt(ua, T_("All Volumes from all Pools")); /* 13 */
+    AddPrompt(ua, T_("Enabled")),                    /* 14 */
+        AddPrompt(ua, T_("RecyclePool")),            /* 15 */
+        AddPrompt(ua, T_("Action On Purge")),        /* 16 */
+        AddPrompt(ua, T_("Storage")),                /* 17 */
+        AddPrompt(ua, T_("Done"));                   /* 18 */
```
Comment (sebsura) [inline core/src/dird/ua_update.cc:637]:
```suggestion
    AddPrompt(ua, T_("Enabled"));                    /* 14 */
    AddPrompt(ua, T_("RecyclePool"));                /* 15 */
    AddPrompt(ua, T_("Action On Purge"));            /* 16 */
    AddPrompt(ua, T_("Storage"));                    /* 17 */
    AddPrompt(ua, T_("Done"));                       /* 18 */
```
Klappt zwar auch so wie es war, sieht aber komisch aus.

File: core/src/lib/ini.cc [hunk 2]
```diff
@@ -225,22 +225,22 @@ static void s_err(const char* file, int line, LEX* lc, const char* msg, ...)
...
 
     //   } else if (ini->ctx) {       /* called from plugin */
     //      ini->bareos_core_functions->JobMessage(ini->ctx, __FILE__, __LINE__,
     //      M_FATAL, 0,
-    //                    _("Config file error: %s\n"
+    //                    T_("Config file error: %s\n"
     //                      "            : Line %d, col %d of file %s\n%s\n"),
     //                            buf.c_str(), lc->line_no, lc->col_no,
     //                            lc->fname, lc->line);
     //
```
Comment (sebsura) [inline core/src/lib/ini.cc:239]: delete if not needed

File: core/src/include/baconfig.h [hunk 3]
```diff
@@ -85,34 +85,24 @@ void InitWinAPIWrapper();
...
-#    define textdomain(d)
-#  endif
-#  ifndef bindtextdomain
-#    define bindtextdomain(p, d)
-#  endif
-#  ifndef setlocale
-#    define setlocale(p, d)
-#  endif
+#  define T_(s) (s)
+#  define N_(s) (s)
```
Comment (sebsura) [inline core/src/include/baconfig.h:102]: What is the difference between `N_()` and `NT_()` ? Should we unify those to just `NT_()` ?

File: core/src/lib/ini.cc [hunk 4]
```diff
@@ -270,22 +270,22 @@ static void s_warn(const char* file, int line, LEX* lc, const char* msg, ...)
...
 
     //   } else if (ini->ctx) {       /* called from plugin */
     //      ini->bareos_core_functions->JobMessage(ini->ctx, __FILE__, __LINE__,
     //      M_WARNING, 0,
-    //                    _("Config file warning: %s\n"
+    //                    T_("Config file warning: %s\n"
     //                      "            : Line %d, col %d of file %s\n%s\n"),
     //                            buf.c_str(), lc->line_no, lc->col_no,
     //                            lc->fname, lc->line);
     //
```
Comment (sebsura) [inline core/src/lib/ini.cc:284]: Same

File: core/src/lib/tls_openssl.cc [hunk 5]
```diff
@@ -317,7 +317,7 @@ void TlsOpenSsl::TlsBsockShutdown(BareosSocket* bsock)
 
   if (jcr && jcr->is_passive_client_connection_probing) { return; }
 
-  std::string message{_("TLS shutdown failure.")};
+  std::string message{T_("TLS shutdown failure.")};
```
Comment (sebsura) [inline core/src/lib/tls_openssl.cc:320]:
```suggestion
  const char* message = T_("TLS shutdown failure.");
```
This does not need to be changed, but feel free to do so.

File: core/src/lib/var.cc [hunk 6]
```diff
@@ -2482,79 +2482,80 @@ var_rc_t var_format(var_t* var,
...
-                                                 */
-    _("offsets in cut operation delimited by unknown character"), /* VAR_ERR_INVALID_OFFSET_DELIMITER
-                                                                   */
-    _("range out of bounds in cut operation"), /* VAR_ERR_RANGE_OUT_OF_BOUNDS */
-    _("offset out of bounds in cut operation"), /* VAR_ERR_OFFSET_OUT_OF_BOUNDS
+    T_("everything ok"),                /* VAR_OK = 0 */
+    T_("incomplete named character"),   /* VAR_ERR_INCOMPLETE_NAMED_CHARACTER */
+    T_("incomplete hexadecimal value"), /* VAR_ERR_INCOMPLETE_HEX */
+    T_("invalid hexadecimal value"),    /* VAR_ERR_INVALID_HEX */
+    T_("octal value too large"),        /* VAR_ERR_OCTAL_TOO_LARGE */
```
Comment (sebsura) [inline core/src/lib/var.cc:2489]: For arrays such as this I much prefer the C syntax:
```suggestion
    [VAR_OK] = T_("everything ok"),
    [VAR_ERR_INCOMPLETE_NAMED_CHARACTER] = T_("incomplete named character"),
    [VAR_ERR_INCOMPLETE_HEX] = T_("incomplete hexadecimal value"),
    [VAR_ERR_INVALID_HEX] = T_("invalid hexadecimal value"),
    [VAR_ERR_OCTAL_TOO_LARGE] = T_("octal value too large"),
```
even if not c++ standard compliant.  But this change does not belong in this pr.

**Review by sebsura — CHANGES_REQUESTED** [review]
Looks good.  I added some suggestions, tell me what you think of them.

**Comment by arogge** [comment]
I had to rebase and force-push to fix some conflicts. Sorry for that.

**Review by sebsura — CHANGES_REQUESTED** [review]
You cannot remove `ini.h` and `ini.cc` since they are actually required by `test-plugin-fd.cc`.

If that file is not used anymore, we should probably delete it as well.

**Comment by arogge** [comment]
That test plugin looks like something that bitrotted away. At least there is no build configuration on Jenkins that would even try to build it.
I'll see if we should re-enable or remove it.

**Review by sebsura — APPROVED** [review]
Good work!  You just need to fix the build issues but Ill already approve of it for now.

**Pull request merged**